pkg/apis/serving/k8s_validation.go

```diff
@@ -451,6 +451,10 @@ func validateReadinessProbe(p *corev1.Probe) *apis.FieldError {
 		errs = errs.Also(apis.ErrOutOfBoundsValue(p.PeriodSeconds, 0, math.MaxInt32, "periodSeconds"))
 	}
 
+	if p.InitialDelaySeconds < 0 {
+		errs = errs.Also(apis.ErrOutOfBoundsValue(p.InitialDelaySeconds, 0, math.MaxInt32, "initialDelaySeconds"))
```

Comment:
Do we really want 2 billion second even theoretical delay?

Reply:
Actually we don't need 
Initially i have handled like below
`apis.ErrInvalidValue(strconv.Itoa(int(p.InitialDelaySeconds))+": must be greater than or equal to 0", "initialDelaySeconds")`
Please Let me know opinion on this

Reply:
Lets keep this for consistency with the other validations on the probe.